client/components/UserSettings/ConfigureUserSettingsModal.tsx

```diff
@@ -29,11 +36,14 @@ import { CountrySelect } from './CountrySelect';
 
 export const ConfigureUserSettingsModal: React.FC<ModalProps> = ({ show, onClose }) => {
     const initialRef = useRef(null);
-    const { user, updateUserSettings } = useUserContext();
+    const { user, updateUserSettings, logoutUser, deleteUser } = useUserContext();
     const {
         loading,
         execute: updateSettings
     } = useAction(updateUserSettings, { onSuccess: onClose });
+    const {
+        loading: loadingDelete
+    } = useAction(deleteUser, { onSuccess: onClose });
```

Comment:
```suggestion
    const {
        loading: loadingDelete,
        execute: deleteAccount
    } = useAction(deleteUser, { onSuccess: onClose });
```
For the loading value to be set correctly, you have to use the `execute` method provided by the hook. So please use `deleteAccount` instead of `deleteUser` in `onAccountDelete` below